src/main/java/com/SAS/User/User.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * The class represents a user in the system
+ */
+
+package com.SAS.User;
+
+public abstract class User {
+
```

Comment:
User is an Abstract class and you send it as parameter to alot of Constructors, you won't be able to initiate any of the classes you send User to as Parameter because you cant type:
`User x = new User()`

please send the Parameters that needed to initiate User instead of User it self,
and then inside the Constructor of the inhearited classes, use:
`super()` constructor

Reply:
It's according to the decorator design pattern.